axelor-account/src/main/java/com/axelor/apps/account/service/move/attributes/MoveAttrsServiceImpl.java

```diff
@@ -423,4 +432,33 @@ protected boolean isPartnerRequired(Journal journal) {
             || journal.getJournalType().getTechnicalTypeSelect()
                 == JournalTypeRepository.TECHNICAL_TYPE_SELECT_SALE);
   }
+
+  @Override
+  public void addCompanyDomain(Move move, Map<String, Map<String, Object>> attrsMap) {
+    List<Company> companyList = companyRepository.all().fetch();
+    List<Company> companyWithAccountConfig = new ArrayList<>();
+
+    if (ObjectUtils.notEmpty(companyList)) {
+      for (Company company : companyList) {
+        try {
+          accountConfigService.getAccountConfig(company);
+          companyWithAccountConfig.add(company);
+        } catch (AxelorException e) {
+          TraceBackService.trace(e);
+        }
+      }
+    }
+
+    String companyIds =
+        CollectionUtils.isEmpty(companyList)
+            ? "0"
+            : companyWithAccountConfig.stream()
+                .map(Company::getId)
+                .map(Objects::toString)
+                .collect(Collectors.joining(","));
+
+    String domain = String.format("self.id IN (%s)", companyIds);
+
+    this.addAttr("company", "domain", domain, attrsMap);
```

Comment:
I don't think it should be done like this, you could just do something like `companyRepository.all().filter("self.accountConfig IS NOT NULL").fetch()`

Reply:
Resolved